### PR TITLE
Add table oci_core_image_custom. Closes #273

### LINKED ISSUE #273: Add table oci_core_image_custom
**References**
Please refer oci_core_image table.

### PR DESCRIPTION
**Pull request by bigdatasourav**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
SETUP: tests/oci_core_image_custom []

PRETEST: tests/oci_core_image_custom

TEST: tests/oci_core_image_custom
Running terraform
oci_core_vcn.named_test_resource: Creating...
oci_core_vcn.named_test_resource: Creation complete after 1s [id=ocid1.vcn.oc1.ap-mumbai-1.amaaaaaa6igdexaapmd7ju6eajbiouzvf2lhziyyhljhjj5smv5ukgvxac4a]
oci_core_subnet.named_test_resource: Creating...
oci_core_subnet.named_test_resource: Creation complete after 3s [id=ocid1.subnet.oc1.ap-mumbai-1.aaaaaaaabkcjcwjnzf3oi2llluexaeegw3lsejibo4nveswkautowikm4ftq]
null_resource.test_image: Creating...
null_resource.test_image: Provisioning with 'local-exec'...
null_resource.test_image (local-exec): Executing: ["/bin/sh" "-c" "oci compute image list --compartment-id ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq --all --display-name Oracle-Autonomous-Linux-7.9-2021.05-0 --output json > /private/var/folders/d8/gsp6f_cn3359xnt_gs3cts_80000gn/T/tests/oci_core_image_custom/terraform/test/image.json"]
null_resource.test_image (local-exec): WARNING: Permissions on /Users/sourav/-04-08-07-40.pem are too open.
null_resource.test_image (local-exec): To fix this please try executing the following command:
null_resource.test_image (local-exec): oci setup repair-file-permissions --file /Users/sourav/-04-08-07-40.pem
null_resource.test_image (local-exec): Alternatively to hide this warning, you may set the environment variable, OCI_CLI_SUPPRESS_FILE_PERMISSIONS_WARNING:
null_resource.test_image (local-exec): export OCI_CLI_SUPPRESS_FILE_PERMISSIONS_WARNING=True

null_resource.test_image: Creation complete after 4s [id=7466752822356109888]
data.local_file.image: Reading...
data.local_file.image: Read complete after 0s [id=af3f36599699397c30d26437f50259c029bbd3bf]
null_resource.named_test_resource: Creating...
null_resource.named_test_resource: Provisioning with 'local-exec'...
null_resource.named_test_resource (local-exec): Executing: ["/bin/sh" "-c" "oci compute instance launch --availability-domain TvRS:AP-MUMBAI-1-AD-1 --compartment-id ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq --shape VM.Standard2.1 --subnet-id ocid1.subnet.oc1.ap-mumbai-1.aaaaaaaabkcjcwjnzf3oi2llluexaeegw3lsejibo4nveswkautowikm4ftq --image-id ocid1.image.oc1.ap-mumbai-1.aaaaaaaahkwyrvis5iyrezoric5zmrld4r6xhilonvduzhas25vmyffhirxq --output json > /private/var/folders/d8/gsp6f_cn3359xnt_gs3cts_80000gn/T/tests/oci_core_image_custom/terraform/test/output.json"]
null_resource.named_test_resource (local-exec): WARNING: Permissions on /Users/sourav/-04-08-07-40.pem are too open.
null_resource.named_test_resource (local-exec): To fix this please try executing the following command:
null_resource.named_test_resource (local-exec): oci setup repair-file-permissions --file /Users/sourav/-04-08-07-40.pem
null_resource.named_test_resource (local-exec): Alternatively to hide this warning, you may set the environment variable, OCI_CLI_SUPPRESS_FILE_PERMISSIONS_WARNING:
null_resource.named_test_resource (local-exec): export OCI_CLI_SUPPRESS_FILE_PERMISSIONS_WARNING=True

null_resource.named_test_resource: Provisioning with 'local-exec'...
null_resource.named_test_resource (local-exec): Executing: ["/bin/sh" "-c" "sleep 150"]
null_resource.named_test_resource: Still creating... [10s elapsed]
null_resource.named_test_resource: Still creating... [20s elapsed]
null_resource.named_test_resource: Still creating... [30s elapsed]
null_resource.named_test_resource: Still creating... [40s elapsed]
null_resource.named_test_resource: Still creating... [50s elapsed]
null_resource.named_test_resource: Still creating... [1m0s elapsed]
null_resource.named_test_resource: Still creating... [1m10s elapsed]
null_resource.named_test_resource: Still creating... [1m20s elapsed]
null_resource.named_test_resource: Still creating... [1m30s elapsed]
null_resource.named_test_resource: Still creating... [1m40s elapsed]
null_resource.named_test_resource: Still creating... [1m50s elapsed]
null_resource.named_test_resource: Still creating... [2m0s elapsed]
null_resource.named_test_resource: Still creating... [2m10s elapsed]
null_resource.named_test_resource: Still creating... [2m20s elapsed]
null_resource.named_test_resource: Still creating... [2m30s elapsed]
null_resource.named_test_resource: Creation complete after 2m32s [id=7205681312577788112]
data.local_file.instance: Reading...
data.local_file.instance: Read complete after 0s [id=3e7234c036f50cdede0b36917ad1dba3ade433ec]
oci_core_image.named_test_resource: Creating...
oci_core_image.named_test_resource: Still creating... [10s elapsed]
oci_core_image.named_test_resource: Still creating... [20s elapsed]
oci_core_image.named_test_resource: Still creating... [30s elapsed]
oci_core_image.named_test_resource: Still creating... [40s elapsed]
oci_core_image.named_test_resource: Still creating... [50s elapsed]
oci_core_image.named_test_resource: Still creating... [1m0s elapsed]
oci_core_image.named_test_resource: Still creating... [1m10s elapsed]
oci_core_image.named_test_resource: Still creating... [1m20s elapsed]
oci_core_image.named_test_resource: Still creating... [1m30s elapsed]
oci_core_image.named_test_resource: Still creating... [1m40s elapsed]
oci_core_image.named_test_resource: Still creating... [1m50s elapsed]
oci_core_image.named_test_resource: Still creating... [2m0s elapsed]
oci_core_image.named_test_resource: Still creating... [2m10s elapsed]
oci_core_image.named_test_resource: Still creating... [2m20s elapsed]
oci_core_image.named_test_resource: Still creating... [2m30s elapsed]
oci_core_image.named_test_resource: Still creating... [2m40s elapsed]
oci_core_image.named_test_resource: Still creating... [2m50s elapsed]
oci_core_image.named_test_resource: Still creating... [3m0s elapsed]
oci_core_image.named_test_resource: Still creating... [3m10s elapsed]
oci_core_image.named_test_resource: Still creating... [3m20s elapsed]
oci_core_image.named_test_resource: Creation complete after 3m24s [id=ocid1.image.oc1.ap-mumbai-1.aaaaaaaa4m463xp7cygiq34wmivnwalkdn24k23w3fvfqlmvctwpsehpvcca]
null_resource.destroy_test_resource: Creating...
null_resource.destroy_test_resource: Provisioning with 'local-exec'...
null_resource.destroy_test_resource (local-exec): Executing: ["/bin/sh" "-c" "oci compute instance terminate --instance-id ocid1.instance.oc1.ap-mumbai-1.anrg6ljr6igdexac6nqqn63kaps54qd5jerczk3eql3h4whosnjqofzk2n6a --force"]
null_resource.destroy_test_resource (local-exec): WARNING: Permissions on /Users/sourav/-04-08-07-40.pem are too open.
null_resource.destroy_test_resource (local-exec): To fix this please try executing the following command:
null_resource.destroy_test_resource (local-exec): oci setup repair-file-permissions --file /Users/sourav/-04-08-07-40.pem
null_resource.destroy_test_resource (local-exec): Alternatively to hide this warning, you may set the environment variable, OCI_CLI_SUPPRESS_FILE_PERMISSIONS_WARNING:
null_resource.destroy_test_resource (local-exec): export OCI_CLI_SUPPRESS_FILE_PERMISSIONS_WARNING=True

null_resource.destroy_test_resource: Creation complete after 3s [id=7628639054256057448]

Apply complete! Resources: 6 added, 0 changed, 0 destroyed.

Outputs:

lifecycle_state = "AVAILABLE"
operating_system = "Oracle Autonomous Linux"
operating_system_version = "7.9"
resource_id = "ocid1.image.oc1.ap-mumbai-1.aaaaaaaa4m463xp7cygiq34wmivnwalkdn24k23w3fvfqlmvctwpsehpvcca"
resource_name = "steampipetest1277"
tenancy_ocid = "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq"

Running SQL query: test-get-query.sql

[
  {
    "display_name": "steampipetest1277",
    "id": "ocid1.image.oc1.ap-mumbai-1.aaaaaaaa4m463xp7cygiq34wmivnwalkdn24k23w3fvfqlmvctwpsehpvcca",
    "lifecycle_state": "AVAILABLE"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql

[
  {
    "display_name": "steampipetest1277",
    "id": "ocid1.image.oc1.ap-mumbai-1.aaaaaaaa4m463xp7cygiq34wmivnwalkdn24k23w3fvfqlmvctwpsehpvcca",
    "operating_system": "Oracle Autonomous Linux",
    "operating_system_version": "7.9"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql

null
✔ PASSED

Running SQL query: test-turbot-query.sql

[
  {
    "tenant_id": "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq",
    "title": "steampipetest1277"
  }
]
✔ PASSED

POSTTEST: tests/oci_core_image_custom
Running terraform

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

TEARDOWN: tests/oci_core_image_custom


SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
> select * from oci_core_image_custom where id = 'ocid1.image.oc1.ap-mumbai-1.aaaaaaaadqcilpiqsly5vrynwh6upeemodoa3cgtvyjqq2bogbf5hrniyjva'
+--------------+------------------------------------------------------------------------------------------+-----------------+---------------------+---------------------------------------------------------
| display_name | id                                                                                       | lifecycle_state | time_created        | base_image_id                                           
+--------------+------------------------------------------------------------------------------------------+-----------------+---------------------+---------------------------------------------------------
| test@12      | ocid1.image.oc1.ap-mumbai-1.aaaaaaaadqcilpiqsly5vrynwh6upeemodoa3cgtvyjqq2bogbf5hrniyjva | AVAILABLE       | 2021-08-16 07:16:01 | ocid1.image.oc1.ap-mumbai-1.aaaaaaaalsm2sac6m57q5uloutt2
+--------------+------------------------------------------------------------------------------------------+-----------------+---------------------+---------------------------------------------------------
> select * from oci_core_image_custom
+--------------+------------------------------------------------------------------------------------------+-----------------+---------------------+---------------------------------------------------------
| display_name | id                                                                                       | lifecycle_state | time_created        | base_image_id                                           
+--------------+------------------------------------------------------------------------------------------+-----------------+---------------------+---------------------------------------------------------
| test@12      | ocid1.image.oc1.ap-mumbai-1.aaaaaaaadqcilpiqsly5vrynwh6upeemodoa3cgtvyjqq2bogbf5hrniyjva | AVAILABLE       | 2021-08-16 07:16:01 | ocid1.image.oc1.ap-mumbai-1.aaaaaaaalsm2sac6m57q5uloutt2
+--------------+------------------------------------------------------------------------------------------+-----------------+---------------------+---------------------------------------------------------

> select
  display_name,
  id,
  size_in_mbs,
  tags,
  lifecycle_state,
  operating_system
from
  oci_core_image_custom;
+--------------+------------------------------------------------------------------------------------------+-------------+------+-----------------+------------------+
| display_name | id                                                                                       | size_in_mbs | tags | lifecycle_state | operating_system |
+--------------+------------------------------------------------------------------------------------------+-------------+------+-----------------+------------------+
| test@12      | ocid1.image.oc1.ap-mumbai-1.aaaaaaaadqcilpiqsly5vrynwh6upeemodoa3cgtvyjqq2bogbf5hrniyjva | 47694       | {}   | AVAILABLE       | Oracle Linux     |
+--------------+------------------------------------------------------------------------------------------+-------------+------+-----------------+------------------+
> select
  display_name,
  id,
  launch_options ->> 'isPvEncryptionInTransitEnabled' as is_encryption_in_transit_enabled
from
  oci_core_image_custom
where
  launch_options ->> 'isPvEncryptionInTransitEnabled' = 'false';
+--------------+----+----------------------------------+
| display_name | id | is_encryption_in_transit_enabled |
+--------------+----+----------------------------------+
+--------------+----+----------------------------------+
> select
  display_name,
  id,
  launch_options ->> 'isPvEncryptionInTransitEnabled' as is_encryption_in_transit_enabled
from
  oci_core_image_custom
where
  launch_options ->> 'isPvEncryptionInTransitEnabled' = 'true';
+--------------+------------------------------------------------------------------------------------------+----------------------------------+
| display_name | id                                                                                       | is_encryption_in_transit_enabled |
+--------------+------------------------------------------------------------------------------------------+----------------------------------+
| test@12      | ocid1.image.oc1.ap-mumbai-1.aaaaaaaadqcilpiqsly5vrynwh6upeemodoa3cgtvyjqq2bogbf5hrniyjva | true                             |
+--------------+------------------------------------------------------------------------------------------+----------------------------------+
```
</details>

### CODE REVIEW

File: oci/table_oci_core_image_custom.go [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+	"github.com/turbot/steampipe-plugin-sdk/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/plugin/transform"
+)
+
+//// TABLE DEFINITION
+
+func tableCoreImageCustom(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "oci_core_image_custom",
+		Description: "OCI Core Image_custom",
```
Comment (khushboo9024) [inline oci/table_oci_core_image_custom.go:20]:
```suggestion
		Description: "OCI Core Image Custom",
```

File: docs/tables/oci_core_image_custom.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+```sql
+select
+  display_name,
+  id,
+  launch_options ->> 'isPvEncryptionInTransitEnabled' as is_encryption_in_transit_enabled
+from
+  oci_core_image_custom
+where
+  launch_options ->> 'isPvEncryptionInTransitEnabled'  = 'false';
```
Comment (khushboo9024) [inline docs/tables/oci_core_image_custom.md:31]:
```suggestion
  launch_options ->> 'isPvEncryptionInTransitEnabled' = 'false';
```

File: docs/tables/oci_core_image_custom.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
+# Table: oci_core_image_custom
+
+An image is a template of a virtual hard drive. The image determines the operating system and other software for an instance.
```
Comment (khushboo9024) [inline docs/tables/oci_core_image_custom.md:3]: Description in not proper, it should be about custom images, we have the same description in table `oci_core_image`, please check.

**Review by khushboo9024 — CHANGES_REQUESTED** [review]
Please add the example query result in logs.

**Review by khushboo9024 — APPROVED** [review]

**Review by rajlearner17 — APPROVED** [review]
LGTM

**Pull request merged**